cupboard_app/views.py

```diff
@@ -166,6 +171,100 @@ def get(self, request: Request) -> Response:
         return Response(serializer.data)
 
 
+@extend_schema(tags=['List_Item'])
+class ListItemAPIView(APIView):
+    missingMsg = "Required value misssing from sent request,"
+    missingMsg += "please ensure all items are sent in the following format:"
+    missingMsg += "{\n  username: [USERNAME],\n  listName: [LISTNAME],\n  ingredient: "
+    missingMsg += "[INGREDIENT],\n  amount: [AMOUNT/QUANTITY],\n  unit: [MEASURMENT UNIT]\n}"
+
+    @extend_schema(
+        request=None,
+        responses={
+            200: OpenApiResponse(
+                response=MessageSerializer,
+                examples=[
+                    OpenApiExample(
+                        name='Item Updated',
+                        value={'message': UPDATE_SUCCESS_MSG},
+                        status_codes=[200]
+                    ),
+                    OpenApiExample(
+                        name='Referenced Value Does Not Exist',
+                        value={'message': DOES_NOT_EXIST_MSG},
+                        status_codes=[200]
+                    ),
+                    OpenApiExample(
+                        name='Ingredient Does Not Exist',
+                        value={'message': f'{UPDATE_FAILED_MSG} Ingredient does not exist.'},
+                        status_codes=[200]
+                    ),
+                ]
+            ),
+            401: auth_failed_response,
+            500: OpenApiResponse(
+                response=MessageSerializer,
+                examples=[
+                    OpenApiExample(
+                        name='Required Value Missing',
+                        value={'message': missingMsg},
+                        status_codes=[500]
+                    )
+                ]
+            ),
+        }
+    )
+    def post(self, request: Request) -> Response:
+        """
+        adds an ingredient to a list
+
+        Args:
+            request: The rest framework Request post object containing json in the following format
+            {
+                username: [USERNAME],
+                listName: [LISTNAME],
+                ingredient: [INGREDIENT],
+                amount: [AMOUNT/QUANTITY],
+                unit: [MEASURMENT UNIT]
+            }
+
+        Returns:
+            A json object with the result of the operation.
+            Output Format:
+            {
+                "message": [RESULT MSG]
+            }
+        """
+        missingMsg = "Required value misssing from sent request,"
+        missingMsg += "please ensure all items are sent in the following format:"
+        missingMsg += "{\n  username: [USERNAME],\n  listName: [LISTNAME],\n  ingredient: "
+        missingMsg += "[INGREDIENT],\n  amount: [AMOUNT/QUANTITY],\n  unit: [MEASURMENT UNIT]\n}"
+        body = json.loads(request.body)
```

Comment:
Use request.data instead then you won't need the json library because request.data returns a fully parsed python dictionary.